### PR TITLE
ZOSVPC - 2871 Implementation of giving an option of creating z/os vsi with a specified profile size for standard variation.

### PR DESCRIPTION
**Pull request by Nikhil22February**
1. Give the option of creating a z/OS VSI
with a specified profile size
validate the input profile size to ensure that z/OS VSI will IPL successfully - that is, >= 2 vCPUS and >= 16GB RAM. For example, the following existing IBM Z profiles will NOT work:
bz2-1x4
bz2-2x8
cz2-2x4
cz2-4x8
2. Support the profile size option for standard variations

### CODE REVIEW

File: solutions/standard/variables.tf [hunk 1]
```diff
@@ -35,6 +35,15 @@ variable "ssh_public_key" {
   }
 }
 
+variable "machine_type" {
+  type = string 
+  description = "input machine type: valid values are: bz2-4x16, bz2-8x32, bz2-16x64, cz2-8x16, cz2-16x32, mz2-2x16, mz2-4x32, mz2-8x64, mz2-16x128"
+  validation {
+    condition  = contains(["mz2o-2x16","bz2-4x16", "bz2-8x32", "bz2-16x64", "cz2-8x16", "cz2-16x32", "mz2-2x16", "mz2-4x32", "mz2-8x64", "mz2-16x128"], var.machine_type)
+    error_message = "Valid values for machine_type are: mz2o-2x16, bz2-4x16, bz2-8x32, bz2-16x64"
```
Comment (kevinkuhner) [inline solutions/standard/variables.tf:43]: Update your error message to contain all the same valid machine type values as in the description. Also - remove "mz2o-2x16" as that is an internal profile and most accounts will not have.

**Review by kevinkuhner — CHANGES_REQUESTED** [review]
@Nikhil22February - 1 comment for this PR.

File: solutions/standard/variables.tf (hunk 1 above)
Comment (Nikhil22February) [reply solutions/standard/variables.tf]: Made changes as suggested.

**Review by kevinkuhner — APPROVED** [review]
@Nikhil22February  - I this is ok now, but I noticed that you removed `mz2o-2x16` from the "contains()" check.  This means that any internal account with this special profile enabled will not pass the check and so the user would need to manually update the check for this case.  That may be OK... I'm just pointing it out.   Another option might be to leave this special profile in the contains check but just not include it in the description or error messages.

**Comment by Nikhil22February** [comment]
> @Nikhil22February - I this is ok now, but I noticed that you removed `mz2o-2x16` from the "contains()" check. This means that any internal account with this special profile enabled will not pass the check and so the user would need to manually update the check for this case. That may be OK... I'm just pointing it out. Another option might be to leave this special profile in the contains check but just not include it in the description or error messages.

Thank you for pointing this out. I have added the profile size mz2o-2x16 in contains check of machine_type variable.

**Pull request merged**